id-library-showcase/src/main.rs

```diff
@@ -355,64 +440,153 @@ async fn handle_verify_claim(vc: VerifyClaim) -> anyhow::Result<()> {
     Ok(())
 }
 
-fn handle_prove_attribute_in_range(pair: ProveAttributeInRange) {
-    let randomness: PedersenRandomness<ExampleCurve> = match read_json_from_file(pair.randomness) {
-        Ok(r) => r,
-        Err(e) => {
-            eprintln!("Could not parse randomness: {}. Terminating.", e);
-            return;
-        }
-    };
-
-    let global_ctx = {
-        if let Some(gc) = read_global_context(pair.global) {
-            gc
-        } else {
-            eprintln!("Cannot read global context information database. Terminating.");
-            return;
-        }
-    };
+async fn handle_prove_attribute_in_range(pair: ProveAttributeInRange) -> anyhow::Result<()> {
+    let account = pair.account;
+    let attribute_tag = pair.attribute_tag;
+    let mut client = endpoints::Client::connect(pair.endpoint, "rpcadmin".to_string()).await?;
+    let consensus_info = client.get_consensus_status().await?;
+    let global_ctx = client
+        .get_cryptographic_parameters(&consensus_info.last_finalized_block)
+        .await?;
+    // let randomness: PedersenRandomness<ExampleCurve> = read_json_from_file(pair.randomness)?;
+    let (attribute, randomness): (AttributeKind, PedersenRandomness<ExampleCurve>) =
+        match (pair.attribute, pair.randomness, pair.wallet) {
+            (Some(attribute), Some(randomness_file), _) => 
+            match read_json_from_file(randomness_file) {
+                Ok(r) => (attribute, r),
+                Err(e) => {
+                    anyhow::bail!("Could not parse randomness: {}. Terminating.", e)
+                }
+            },
```

Comment:
Recommend instead to do
```rust
let r = read_json_from_file(randomness_file).context("Could not parse randomness.")?
```
instead of the match statement.